Sources/ArgumentParser/Parsing/CommandParser.swift

```diff
@@ -456,16 +456,32 @@ extension CommandParser {
       }
 
       guard
-        let s = args.popFirst(),
-        let cursorIndexWithinCompletingArgument = Int(s)
+        let arg = args.popFirst(),
+        let cursorIndexWithinCompletingArgument = Int(arg)
       else {
         throw ParserError.invalidState
       }
 
+      let completingPrefix: String
+      if let completingArgument = args.last {
+        completingPrefix = String(
+          completingArgument.prefix(
+            upTo: completingArgument.index(
+              completingArgument.startIndex,
+              offsetBy: cursorIndexWithinCompletingArgument
+            )
+          )
```

Comment:
There's a prefix overload that just takes the count of elements to include, so this can be simplified to `.prefix(cursorIndexWithinCompletingArgument)`.

Reply:
I will update the PR now. I'm still new to Swift, so my API knowledge is limited. I didn't even realize how verbose it is to get substrings in Swift. Thanks for pointing this out. All similar tips will be greatly appreciated, as it will produce much nicer code & get me up to speed much faster. Thanks.

Reply:
@natecook1000 Pushed the fix. Thanks again for the tip.